src/bsoncxx/view_or_value.hpp

```diff
@@ -147,6 +163,30 @@ BSONCXX_INLINE bool operator!=(View lhs, const view_or_value<View, Value>& rhs)
     return !(rhs == lhs);
 }
 
+///
+/// Equality operators for comparison with plain Values.
+///
+
+template <typename View, typename Value>
+BSONCXX_INLINE bool operator==(const view_or_value<View, Value>& lhs, Value rhs) {
```

Comment:
Definitely const& for the Values here, otherwise you are making a copy of something with ownership